File: js/src/program/ops/opString.js
export default {
  /** { string: string } */
  op(runtime, input, params, scope, next) {
    return next(params.string ?? '', scope);
  },

  /** { string: string } */
  map(runtime, params) {
    return {
      string: runtime.assertType(params.string, 'string'),
    };
  },
};

File: go/program/ops/opString.go
package ops

import (
	"github.com/2manyvcos/jpl/go/definition"
	"github.com/2manyvcos/jpl/go/jpl"
)

type opString struct{}

// { string: string }
func (opString) OP(runtime jpl.JPLRuntime, input any, params definition.JPLInstructionParams, scope jpl.JPLRuntimeScope, next jpl.JPLScopedPiper) ([]any, jpl.JPLError) {
	return next.Pipe(params.String, scope)
}

// { string: string }
func (opString) Map(runtime jpl.JPLRuntime, params jpl.JPLInstructionParams) (definition.JPLInstructionParams, jpl.JPLError) {
	return definition.JPLInstructionParams{
		String: params.String,
	}, nil
}
